### DISCUSSION TITLE
Module not found: Can't resolve 'fs'  even using in getStaticPaths and getStaticProps

### DISCUSSION THREAD

**Discussion by boomwaterstep** [vercel/next.js#14450] [Help] [answered]
```
import {
  getAllMoviesTitles,
  getMovieData,
 getVideoDownloadLink
} from "../../lib/movies";
import Head from "next/head";
import Player from "../../components/Player";
import Navbar from "../../components/Navbar";
import cn from "classnames";
import Link from "next/link";

export async function getStaticPaths() {
  const paths = getAllMoviesTitles();
  return {
    paths,
    fallback: false,
  };
}

export async function getStaticProps({ params }) {
  const movieData = await getMovieData(params.title);
  return {
    props: {
      movieData,
    },
  };
}
const Theater = ({ movieData }) => {
  return (
    <>
      ...
    </>
  );
};

export default Theater;

```

../lib/movies.js has fs imported at the top of the file but as I understand if it's in getStaticProps, this should work just fine. Any idea what did I do wrong?

**Comment by boomwaterstep** [comment, marked as answer]
Solved. You can't import anything from anywhere that have fs imported if you aren't gonna use it in getStaticProps or the error will occur. 
In my case **getVideoDownloadLink** is imported but not being used and causes the problem.

**Reply by danfusaro** [reply]
Saved me here, was driving me crazy.

**Reply by CoderJakaria** [reply]
oh thanks dude thats solveeeed

**Reply by varunyn** [reply]
@boomwaterstep that solved the issue, but can you explain more on the cause of error.

**Reply by MoOx** [reply]
See https://github.com/vercel/next.js/issues/16153#issuecomment-673374488: in short Node import & code MUST be in `pages` folder, otherwise not correctly removed for the frontend
